### PR TITLE
SOF-253: Unit tests - Permissions Screen

### PR DESCRIPTION
**Pull request by JohnnieShen**
This PR implements unit tests for the permission screen and the main VM. It covers all code paths and edge cases. To test it you should run both test files on a physical device.

### CODE REVIEW

**Comment by aryamanshodhan** [comment]
@JohnnieShen Can you add these tests to the github workflows?

**Comment by JohnnieShen** [comment]
> @JohnnieShen Can you add these tests to the github workflows?

I believe you don't have to do anything for it to be part of CI, as soon as it is merged it in it will run with the next CI task

**Review by aryamanshodhan — DISMISSED** [review]
Looks good Johnny! Very comprehensive set of tests. Please rerun to make sure they pass before merging. One conflict to address!

**Comment by JohnnieShen** [comment]
Unfortuantely this is gonna have to wait until the permission screen new UI is merged in

File: app/src/main/java/com/vci/vectorcamapp/main/presentation/PermissionScreen.kt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// app/src/main/java/com/vci/vectorcamapp/main/presentation/PermissionScreen.kt
```
Comment (aryamanshodhan) [inline app/src/main/java/com/vci/vectorcamapp/main/presentation/PermissionScreen.kt]: Remove this comment

**Review by aryamanshodhan — DISMISSED** [review]
Looks good Johnny. Just need to remove this one comment inside the PermissionScreen.

**Review by aryamanshodhan — APPROVED** [review]

**Review by ThomasT0830 — APPROVED** [review]
